src/components/Personalization/createExecuteCachedViewDecisions.js

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import isNonEmptyArray from "../../utils/isNonEmptyArray";
+
+export default ({ viewCache, executeViewDecisions, collect }) => {
+  return ({ viewName }) => {
+    const viewDecisions = viewCache.getView(viewName);
+
+    if (isNonEmptyArray(viewDecisions)) {
+      executeViewDecisions(viewDecisions);
+      return;
+    }
+
+    const xdm = { web: { webPageDetails: { viewName } } };
+
+    collect({ meta: {}, xdm });
```

Comment:
no, this behavior is only related to SPA and views.

Reply:
I also see collect logic here though: https://github.com/adobe/alloy/blob/e32c6f8acfaf809fb880ed4bb4ad57646b24ddd3/src/components/Personalization/createExecuteDecisions.js#L49

I'm trying to figure out how what we're doing here in createExecuteCachedViewDecisions.js relates to what we're doing in `createExecuteDecisions`. Anyway, my question isn't very clear because I don't really have a very specific question. I think what I'm really trying to figure out is if we could/should call `executeViewDecisions` from this module _regardless_ of whether the `viewDecisions` array is empty and let `executeViewDecisions` be the only module that deals with `display` notifications. I don't understand `executeViewDecisions` enough to know what that would look like though, so my question was a shot in the dark and quite vague. Don't worry about spending a bunch of time trying to answer. It's probably fine as it is.